### ISSUE TITLE
Return query stats/metadata with execution

### ISSUE THREAD

**Issue by RedCraig** [technige/py2neo#423]
This is similar to #408. 

I would like get the queries stats back upon executing a cypher statement with py2neo.
Stats like relationships created, nodes created, etc:

``` bash
$curl \
-H Accept:application/json \
-H Content-Type:application/json \
http://localhost:7474/db/data/cypher?includeStats=true \
-d '{"query" : "MATCH (n) return count(n)"}' \
{
  "columns" : [ "count(n)" ],
  "data" : [ [ 133733 ] ],
  "stats" : {
    "relationships_created" : 0,
    "nodes_deleted" : 0,
    "relationship_deleted" : 0,
    "indexes_added" : 0,
    "properties_set" : 0,
    "constraints_removed" : 0,
    "indexes_removed" : 0,
    "labels_removed" : 0,
    "constraints_added" : 0,
    "labels_added" : 0,
    "nodes_created" : 0,
    "contains_updates" : false
  }
}
```

_Note:_ I can only find this with the [deprecated legacy HTTP endpoint](http://neo4j.com/docs/2.2.1/rest-api-cypher.html#rest-api-retrieve-query-metadata).

I can't see any mention of it in the docs for the newer [transactional cypher HTTP endpoint](http://neo4j.com/docs/2.2.1/rest-api-transactional.html).
The new endpoint doesn't send any metadata back with `includeStats`:

``` bash
$curl \
> -H Accept:application/json \
> -H Content-Type:application/json \
> http://localhost:7474/db/data/transaction/commit?includeStats=true \
> -d '{"statements" : [{"statement": "MATCH (n) return count(n)"}]}' \
{"results":[{"columns":["count(n)"],"data":[{"row":[133733]}]}],"errors":[]}
```

Hopefully there's a different way to request query stats using the new endpoint.

Thanks!
Craig

**Comment by dmk255** [comment]
So it looks like this was added to Cursors. Is there any way to get the stats for a merge / create action? It would be really helpful to know whether I actually created a node or just asserted that it exists.

Logging does show some stats (`watch(neo4j.bolt)`).

**Comment by dmk255** [comment]
This will likely be addressed in #594
